evm/db/chain.py

```diff
@@ -93,12 +99,14 @@ def __init__(self, db, account_state_class=MainAccountStateDB, trie_class=Hexary
     #
     # Canonical chain API
     #
+    @abstractmethod
     def get_canonical_head(self):
         """
         Returns the current block header at the head of the chain.
         """
         raise NotImplementedError("ChainDB classes must implement this method")
 
+    @abstractmethod
```

Comment:
Note to self/others

[This section of our style guide](https://github.com/pipermerriam/ethereum-dev-tactical-manual/blob/master/style-guide.md#avoid-from-x-import-y-for-most-standard-library-imports) recommends that this should be done as

```
import abc

class Thing:
    @abc.abstractmethod
    def some_method(self):
        ...
```

However, I think this specific use case is a good explicit exception to that style as the extra `abc` prefix makes this less readable in my opinion.  Going to add that to the style guide but I thought I'd highlight it here.

Reply:
Thank you for bringing the style guide to my attention!